base/task.jl

```diff
@@ -105,12 +159,14 @@ function task_done_hook(t::Task)
                 active_repl_backend.in_eval
                 throwto(active_repl_backend.backend_task, result)
             end
-            let bt = catch_backtrace()
-                # run a new task to print the error for us
-                @schedule with_output_color(:red, STDERR) do io
-                    print(io, "ERROR (unhandled task failure): ")
-                    showerror(io, result, bt)
-                    println(io)
+            if !suppress_excp_printing(t)
```

Comment:
I wish we didn't need this. It always feels wrong to have a "don't print" option.